valhalla/mjolnir/ferry_connections.h

```diff
@@ -32,14 +32,18 @@ constexpr uint32_t kPermanent = 1;
 // be "adjacent" to an node that is permanently labeled.
 constexpr uint32_t kTemporary = 2;
 
+constexpr uint32_t kFerryUpClass = static_cast<uint32_t>(baldr::RoadClass::kPrimary);
+
 // NodeLabel - for simple shortest path
 struct NodeLabel {
   float cost;
   uint32_t node_index;
   uint32_t pred_node_index;
+  bool dest_only;
+  bool remove_dest_only;
```

Comment:
two new properties for the label:
- `dest_only` is to keep track 
- `remove_dest_only` is only set if we can be sure we want to remove it eventually